File: 0206_Reverse Linked List/ReverseLinkedList2.cs
public class Solution {
   public ListNode ReverseList(ListNode head) {
        return Reverse(null, head);     
   }

    private ListNode Reverse(ListNode previous, ListNode current)
    {
        if(current == null) return previous;        
        var next = current.next;
        current.next = previous;
        return Reverse(current, next);
    }
}

File: 0206_Reverse Linked List/ReverseLinkedList2.java
class Solution {
    public ListNode reverseList(ListNode head) {
        return reverseList(head, null);
    }
    
    private ListNode reverseList(ListNode curr, ListNode prev)
    {
        if(curr == null) return prev;
        ListNode next = curr.next;
        curr.next = prev;
        return reverseList(next, curr);
    }
}